sentry/src/routes/channel.rs

```diff
@@ -102,6 +104,39 @@ pub async fn last_approved<A: Adapter>(
         .unwrap())
 }
 
+pub async fn insert_events<A: Adapter + 'static>(
+    req: Request<Body>,
+    app: &Application<A>,
+) -> Result<Response<Body>, ResponseError> {
+    let session = req
+        .extensions()
+        .get::<Session>()
+        .expect("request session")
+        .to_owned();
+
+    let channel = req
+        .extensions()
+        .get::<Channel>()
+        .expect("Request should have Channel")
+        .to_owned();
+
+    let into_body = req.into_body();
+    let body = hyper::body::to_bytes(into_body).await?;
+    let request_body = serde_json::from_slice::<HashMap<String, Vec<Event>>>(&body)?;
+    let events = request_body
+        .get("events")
+        .ok_or_else(|| ResponseError::BadRequest("invalid request".to_string()))?;
+
+    app.event_aggregator
+        .record(app, &channel, &session, &events)
```

Comment:
In the JS implementation, to achieve max throughput we get the channel in the `recorder` itself. I suggest you use the same approach. There's no point to get the channel on every request here, when it can be retrieved when constructing the recorder, since it's immutable (even when we add dynamic data, it won't be all that important to update it with every request)